### PR TITLE
Building with clang on Linux

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by chfast** [comment]
Thanks. Are you building standalone version?

**Comment by arkpar** [comment]
No, I've tested it with cpp-ethereum

**Comment by arkpar** [comment]
I'll make sure standalone is working as well

**Comment by arkpar** [comment]
Done

File: libevmjit/Array.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace jit
 {
 
 static const auto c_reallocStep = 1;
-static const auto c_reallocMultipier = 2;
+//static const auto c_reallocMultipier = 2;
```
Comment (arkpar) [inline libevmjit/Array.cpp]: Commented out since the code using it is also commented out

File: libevmjit/Runtime.h [hunk 2]
```diff
@@ -22,7 +22,6 @@ class Runtime
 	Env* m_env = nullptr;			///< Pointer to environment proxy. Expected by compiled contract.
 	byte* m_memData = nullptr;
 	uint64_t m_memSize = 0;
-	uint64_t m_memCap = 0;
```
Comment (chfast) [inline libevmjit/Runtime.h]: That one is actually needed by LLVM on ABI level. Not visible to C++ compiler. Maybe I should mark it with protected access specifier.

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,9 +5,10 @@ project(evmjit)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 set(CMAKE_AUTOMOC OFF)
 
-if(${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-else()
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
 	set(CMAKE_CXX_FLAGS "-std=c++11 -Wall -Wextra -Wconversion -Wno-sign-conversion -Wno-unknown-pragmas")
+elseif((${CMAKE_CXX_COMPILER_ID} MATCHES "Clang") AND (${CMAKE_SYSTEM_NAME} MATCHES "Linux"))
+	set(CMAKE_CXX_FLAGS "-std=c++11 -Wall -Wno-unknown-pragmas -Wextra -Wno-unused-parameter -stdlib=libstdc++ -fcolor-diagnostics -Qunused-arguments")
```
Comment (chfast) [inline CMakeLists.txt]: I think the true problem here is that `CMAKE_CXX_FLAGS` are overridden instead of appended...

**Comment by arkpar** [comment]
I'll make another PR to cpp-ethereum instead

**Pull request closed without merging**